Plugin Check is failing for plugins that depend on other plugins on WP 6.5

Hi @swissspidy,
With the new WP 6.5 dependencies I'm getting an error of a plugin that depends of another.
Current error:
https://github.com/pedro-mendonca/GP-Toolbox/actions/runs/8543241783/job/23406618831
```
Warning: Failed to activate plugin. [Plugin Name] requires 1 plugin to be installed and activated: [plugin-dependency]. .
Error: No plugins activated.
Error: Process completed with exit code 1.
```
And... the test fails.
The current alternative is setting `wp-version` to 6.4, which doesn't rely on the dependencies and the plugin is correctly activated.

Maybe the action could check the plugin dependencies and activate them also, while `wp-cli` doesn't directly support install/activate the required dependencies.

As you've pointed out in https://github.com/swissspidy/wp-plugin-check-action/issues/61#issuecomment-2035419879, this should be fixed upstream on `wp-cli`.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -35,3 +35,9 @@ jobs:
         with:
           build-dir: 'tests/fixtures/hello-dolly-copy'
           wp-version: 'trunk'
+
+      - name: Run plugin check (plugin dependencies)
+        uses: ./
+        continue-on-error: true # Since we expect it to always fail for the test plugin.
+        with:
+          build-dir: 'tests/fixtures/plugin-with-dependencies'

--- a/action.yml
+++ b/action.yml
@@ -89,7 +89,7 @@ runs:
       run: |
         touch .wp-env.json
         > .wp-env.json
-        echo "{ \"core\": $WP_VERSION, \"port\": 8880, \"testsPort\": 8881, \"plugins\": [ \"https://downloads.wordpress.org/plugin/plugin-check.latest-stable.zip\" ], \"mappings\": { \"wp-content/plugins/$PLUGIN_SLUG\": \"$PLUGIN_DIR\" }" >> .wp-env.json
+        echo "{ \"core\": $WP_VERSION, \"port\": 8880, \"testsPort\": 8881, \"plugins\": [ \"https://downloads.wordpress.org/plugin/plugin-check.latest-stable.zip\" ], \"mappings\": { \"wp-content/plugins/$PLUGIN_SLUG\": \"$PLUGIN_DIR\" } }" >> .wp-env.json
 
         npm -g --no-fund i @wordpress/env
         wp-env start --update

--- a/action.yml
+++ b/action.yml
@@ -89,6 +89,10 @@ runs:
         # List all dependencies
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
         DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')
+        
+        echo "Result:"
+        echo $DEPENDENCIES
+        echo "Result over"
 
         if [ -z "$DEPENDENCIES" ]; then
           echo "Plugin has no dependencies"

--- a/action.yml
+++ b/action.yml
@@ -83,6 +83,9 @@ runs:
         CATEGORIES="${CATEGORIES//$'\n'/,}"
         EXCLUDE_DIRS="${EXCLUDE_DIRS//$'\n'/,}"
         ADDITIONAL_ARGS="$CHECKS $EXCLUDE_CHECKS $CATEGORIES $IGNORE_WARNINGS $IGNORE_ERRORS $INCLUDE_EXPERIMENTAL $EXCLUDE_DIRS"
+        
+        # Updating to WP-CLI nightly until 2.11.0 is released so we can do `wp plugin get foo --field=requires_plugins`.
+        wp-env run cli wp cli update --nightly --yes
 
         # List all dependencies
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins

--- a/action.yml
+++ b/action.yml
@@ -105,7 +105,18 @@ runs:
         EXCLUDE_DIRS="${EXCLUDE_DIRS//$'\n'/,}"
         ADDITIONAL_ARGS="$CHECKS $EXCLUDE_CHECKS $CATEGORIES $IGNORE_WARNINGS $IGNORE_ERRORS $INCLUDE_EXPERIMENTAL $EXCLUDE_DIRS"
         
-        # TODO: Install dependencies first, see https://github.com/wordpress/plugin-check-action/issues/66
+        # List all dependencies
+        wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
+        DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')
+
+        if [ -z "$DEPENDENCIES" ];
+          echo "Plugin has no dependencies"
+        else
+          echo "Installing plugin dependencies: $DEPENDENCIES"
+          # Install all dependencies first.
+          wp-env run cli wp plugin install --activate $DEPENDENCIES
+        fi;
+
         wp-env run cli wp plugin activate $PLUGIN_SLUG
 
         # Debugging information

--- a/action.yml
+++ b/action.yml
@@ -90,10 +90,6 @@ runs:
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
         DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')
         
-        echo "Result:"
-        echo $DEPENDENCIES
-        echo "Result over"
-
         if [ -z "$DEPENDENCIES" ]; then
           echo "Plugin has no dependencies"
         else

--- a/action.yml
+++ b/action.yml
@@ -84,9 +84,6 @@ runs:
         EXCLUDE_DIRS="${EXCLUDE_DIRS//$'\n'/,}"
         ADDITIONAL_ARGS="$CHECKS $EXCLUDE_CHECKS $CATEGORIES $IGNORE_WARNINGS $IGNORE_ERRORS $INCLUDE_EXPERIMENTAL $EXCLUDE_DIRS"
         
-        # Updating to WP-CLI nightly until 2.11.0 is released so we can do `wp plugin get foo --field=requires_plugins`.
-        wp-env run cli wp cli update --nightly --yes
-
         # List all dependencies
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
         DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')

--- a/action.yml
+++ b/action.yml
@@ -89,7 +89,7 @@ runs:
       run: |
         touch .wp-env.json
         > .wp-env.json
-        echo "{ \"core\": $WP_VERSION, \"port\": 8880, \"testsPort\": 8881, \"plugins\": [ \"$PLUGIN_DIR\", \"https://downloads.wordpress.org/plugin/plugin-check.latest-stable.zip\" ] }" >> .wp-env.json
+        echo "{ \"core\": $WP_VERSION, \"port\": 8880, \"testsPort\": 8881, \"plugins\": [ \"https://downloads.wordpress.org/plugin/plugin-check.latest-stable.zip\" ], \"mappings\": { \"wp-content/plugins/$PLUGIN_SLUG\": \"$PLUGIN_DIR\" }" >> .wp-env.json
 
         npm -g --no-fund i @wordpress/env
         wp-env start --update

--- a/action.yml
+++ b/action.yml
@@ -60,9 +60,10 @@ runs:
         coverage: none
         tools: composer
 
+# Installing the nightly until 2.11.0 is released so we can do `wp plugin get foo --field=requires_plugins`.
     - name: Install WP-CLI
       run: |
-        curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+        curl -o wp-cli.phar https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar
         chmod +x wp-cli.phar
         mkdir -p bin
         mv wp-cli.phar bin/wp

--- a/action.yml
+++ b/action.yml
@@ -53,27 +53,6 @@ runs:
       with:
         node-version: '20'
 
-    - name: Setup PHP
-      uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231
-      with:
-        php-version: 'latest'
-        coverage: none
-        tools: composer
-
-# Installing the nightly until 2.11.0 is released so we can do `wp plugin get foo --field=requires_plugins`.
-    - name: Install WP-CLI
-      run: |
-        curl -o wp-cli.phar https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar
-        chmod +x wp-cli.phar
-        mkdir -p bin
-        mv wp-cli.phar bin/wp
-        echo "${PWD}/bin" >> $GITHUB_PATH
-      shell: bash
-
-    - name: WP-CLI Info
-      run: wp cli info
-      shell: bash
-
     - name: Set PLUGIN_DIR
       run: |
         PLUGIN_DIR=$(realpath "$BUILD_DIR")
@@ -104,7 +83,7 @@ runs:
         CATEGORIES="${CATEGORIES//$'\n'/,}"
         EXCLUDE_DIRS="${EXCLUDE_DIRS//$'\n'/,}"
         ADDITIONAL_ARGS="$CHECKS $EXCLUDE_CHECKS $CATEGORIES $IGNORE_WARNINGS $IGNORE_ERRORS $INCLUDE_EXPERIMENTAL $EXCLUDE_DIRS"
-        
+
         # List all dependencies
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
         DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')

--- a/action.yml
+++ b/action.yml
@@ -72,6 +72,8 @@ runs:
 
         npm -g --no-fund i @wordpress/env
         wp-env start --update
+        
+        wp-env run cli wp cli info
       shell: bash
       env:
         WP_VERSION: ${{ inputs.wp-version == 'trunk' && '"WordPress/WordPress#master"' || 'null' }}

--- a/action.yml
+++ b/action.yml
@@ -90,7 +90,7 @@ runs:
         wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins
         DEPENDENCIES=$(wp-env run cli wp plugin get $PLUGIN_SLUG --field=requires_plugins | tr ',' ' ')
 
-        if [ -z "$DEPENDENCIES" ];
+        if [ -z "$DEPENDENCIES" ]; then
           echo "Plugin has no dependencies"
         else
           echo "Installing plugin dependencies: $DEPENDENCIES"

--- a/tests/fixtures/plugin-with-dependencies/plugin-with-dependencies.php
+++ b/tests/fixtures/plugin-with-dependencies/plugin-with-dependencies.php
@@ -1,0 +1,8 @@
+<?php
+/*
+Plugin Name: Plugin With Dependencies
+Plugin URI: http://example.com
+Description: This is just a plugin.
+Version: 1.0.0
+Requires Plugins: hello-dolly, performance-lab
+*/

